server/config/service.go

```diff
@@ -87,30 +83,19 @@ func (s *service) reloadMattermostConfig() *model.Config {
 func (s *service) Reconfigure(stored StoredConfig, services ...Configurable) error {
 	mmconf := s.reloadMattermostConfig()
 
-	mattermostSiteURL := mmconf.ServiceSettings.SiteURL
-	if mattermostSiteURL == nil {
-		return errors.New("plugin requires Mattermost Site URL to be set")
+	newConfig := s.GetConfig()
+
+	// GetLicense silently drops an RPC error
+	// (https://github.com/mattermost/mattermost-server/blob/fc75b72bbabf7fabfad24b9e1e4c321ca9b9b7f1/plugin/client_rpc_generated.go#L864).
+	// When running in Mattermost cloud we must not fall back to the on-prem mode, so in case we get a nil retry once.
```

Comment:
we can get nil the second time (and more) as well, no? Is the on-prem mode automatically the non-MattermostCloud mode?

Reply:
Yes, and at this point we assume "Self-managed", see https://github.com/mattermost/mattermost-plugin-apps/blob/181555e6ed4263c6e94a8a4e26706a1bc6192010/server/config/config.go#L124

Reply:
Should we log an error in that case?

Reply:
Suggestion: https://github.com/mattermost/mattermost-plugin-apps/pull/191#discussion_r647406090